src/stream/src/executor/temporal_join.rs

```diff
@@ -59,26 +60,35 @@ pub struct TemporalJoinExecutor<S: StateStore, const T: JoinTypePrimitive> {
 struct TemporalSide<S: StateStore> {
     source: StorageTable<S>,
     table_output_indices: Vec<usize>,
-    cache: ManagedLruCache<OwnedRow, Option<OwnedRow>, DefaultHasher, SharedStatsAlloc<Global>>,
+    cache: ManagedLruCache<OwnedRow, Vec<OwnedRow>, DefaultHasher, SharedStatsAlloc<Global>>,
```

Comment:
I think using `BTreeMap` is more appropriate for updating the cache, just like `HashValueWrapper`.

Reply:
Thanks for the feedback! So the change is for making sure all insert/remove operations when updating cache are O(1) instead of linear scans right? I've updated the code to use a `HashSet` that would serve this purpose in a simpler way. But please let me know if I'm missing something.
A bonus for not using `Vec` is that the generic `EstimateSize` implementation for `Vec` is no longer needed, and I've removed the relevant code.